src/builders.ts

```diff
@@ -90,11 +91,33 @@ export class BomBuilder {
     )
   }
 
-  private fetchNpmLs (projectDir: string, process: NodeJS.Process): any {
+  private getNpmCommand (process: NodeJS.Process): string {
     // `npm_execpath` is used by `npm` internally, and is propagated when kicking `npm run-script`
     /* eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/strict-boolean-expressions -- need to handle optional empty-string */
-    const command = process.env.npm_execpath || 'npm'
+    let command = process.env.npm_execpath || 'npm'
+
+    if (/npx-cli.js/gi.test(command)) {
+      // `npm` must be used for executing `ls`.
+      // Typically `npm-cli.js` is alongside `npx-cli.js`, as such we attempt to use this and validate it exists.
+      const npmPath = command.replace('npx-cli.js', 'npm-cli.js')
+      this.console.debug('DEBUG | command: npx-cli.js usage detected, checking for npm-cli.js ...')
+
+      // If it does not exist, fall back to `npm` as a command.
+      if (existsSync(npmPath)) {
+        this.console.debug('DEBUG | command: npm-cli.js found')
+        command = npmPath
+      } else {
+        // We can't use `npx` to run `ls`, and the `npm-cli.js` could not be found, use `npm`.
+        command = 'npm'
+      }
+    }
 
+    this.console.debug(`DEBUG | command: ${command}`)
+    return command
```

Comment:
:x: dont store the value in a variable to have it eventually returned.

just store a const and return it as a fallaback.
all those modifications to the variable could be transformed to `return` easily,

example (simplified, not actual code):
```javascript
private getNpmCommand (process: NodeJS.Process): string {
  const npm_execpath = process.env.npm_execpath ?? ''

  if ( /npx-foo/.test(npm_execpath)  ) {
    const npm = npm_execpath.replace('npx','npm')
    return fs.existsSycn(npm)
      ? npm  // detected npm
      : 'npm' // fallback to global npm
  }

  return npm_execpath || 'npm'
}

```